pkg/playback/http_replayer.go

```diff
@@ -0,0 +1,198 @@
+package playback
+
+import (
+	"bytes"
+	"encoding/json"
+	"fmt"
+	"io"
+	"net/http"
+	"sync"
+)
+
+// A replayServer is a HTTP server that intercepts incoming requests, and replays recorded responses from the provided cassette.
+type replayServer struct {
+	replayer   *interactionReplayer
+	webhookURL string
+	replayLock *sync.WaitGroup // used to
+}
+
+func newReplayServer(webhookURL string) (httpReplayer replayServer) {
+	httpReplayer = replayServer{}
+	httpReplayer.webhookURL = webhookURL
+	httpReplayer.replayLock = &sync.WaitGroup{}
+	return httpReplayer
+}
+
+func (httpReplayer *replayServer) readCassette(reader io.Reader) error {
+	// TODO: We may want to allow different types of replay matching in the future (instead of simply sequential playback)
+	sequentialComparator := func(req1 interface{}, req2 interface{}) (accept bool, shortCircuitNow bool) {
+		return true, true
+	}
+
+	replayer, err := newInteractionReplayer(reader, httpRequestfromBytes, httpResponsefromBytes, sequentialComparator)
+	if err != nil {
+		return err
+	}
+
+	httpReplayer.replayer = replayer
+
+	return nil
+}
+
+func (httpReplayer *replayServer) handler(w http.ResponseWriter, r *http.Request) {
```

Comment:
here, too -- a docstring for `handler` could go a long way

Reply:
+1

Reply:
Done! 85caf865a9d8b828a9635da735339795656f2173